### PR TITLE
Arnej/more intermediate filters

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst @toregge @havardpe please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.h [hunk 1]
```diff
@@ -103,7 +105,14 @@ class WeakAndBlueprint : public IntermediateBlueprint
 
 //-----------------------------------------------------------------------------
 
-class NearBlueprint : public IntermediateBlueprint
+/** shared implementation for operators that degrade to AND when creating filter for upper-bound case */
+class FilterUpperAndBlueprint : public IntermediateBlueprint
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.h:109]: This is a slippery slope that could lead to templated and/or multiple inheritance real fast. I would prefer utility functions for creating AND/OR filter searches that are called from explicit implementations over this.

**Review by havardpe — COMMENTED** [review]
a bit skeptical about the random mixin, otherwise good

File: searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.h (hunk 1 above)
Comment (geirst) [reply searchlib/src/vespa/searchlib/queryeval/intermediate_blueprints.h:109]: Agree

**Review by geirst — APPROVED** [review]
👍

**Comment by arnej27959** [comment]
@havardpe PTAL

**Review by havardpe — APPROVED** [review]
I see the benefit of adding a createOrFilter utility as well, but that can be done in a later PR

**Pull request merged**